### ISSUE TITLE
Vagrants for getting developers on board

### ISSUE THREAD

**Issue by auroraeosrose** [gophp7/gophp7-ext#8]
We should have a stack of vagrants to help getting new developers on board

We need ones for people doing documentation (http://doc.php.net/ has more on that) 

We need ones for people writing .phpt tests

We need ones for people writing php source code

We should have them in windows (7 and 8.1 and possibly 10) and multiple linux (maybe centos and debiany/ubuntuy)

We should have them in multiple virtual machine backends (hyper-v, virtualbox, vmware at minimum)
so anyone can vagrant up and get started

We also need documentation on how they're built (more than just "this is my box") and how to help out with maintaining these vagrants

Also will possibly need to find a place to host them (we've had offers)

@svpernova09 will be point on this for now - anyone else who is helping out chime in

**Comment by auroraeosrose** [comment]
Oh, and of course basic documentation for "get this up and running"

**Comment by svpernova09** [comment]
@rcousens has a centos box he's working on. Should be a good box to check all things CentOSy stuff on.

**Comment by cebe** [comment]
I would set up some environments using docker for myself and would like to share them too. Should they become part of this repository?

**Comment by lornajane** [comment]
Yes, would love to have the docker ones too - we need some hosting elsewhere, but all the howto and where to find things should go onto our site, please!

**Comment by cebe** [comment]
what kind of hosting do you mean?

**Comment by auroraeosrose** [comment]
@cebe download hosting - afaik docker has stuff in place for that and for vagrant boxes we can use the vagrant cloud - but we may want mirrors or something else??

**Comment by svpernova09** [comment]
IMHO The only hosting we should have to really be concerned about is the Windows boxen. 

Any Linux vagrant boxen we should be able to provision from stock images already hosted elsewhere.

**Comment by auroraeosrose** [comment]
well - check msopentech for some boxes as well - they MIGHT have windows ones?  and you should be able to put the windows ones up on the vagrant cloud, there are some there now :)  they're just all unlicensed demo versions afaik

**Comment by svpernova09** [comment]
AFAIK Vagrant cloud doesn't host the box files for free.

I haven't been able to figure out how to provision Windows boxes. If there is someone who has that nailed down, please get in touch :D

**Comment by cebe** [comment]
In case wee need some hosting to mirror downloads I can provide it. Just ping me if needed.

**Comment by dragoonis** [comment]
@auroraeosrose I'm currently making a php.net mirror vagrant image for website work that I do. Has this been covered on this repo at all ? If not I will carry on and contribute it here.

**Comment by svpernova09** [comment]
@dragoonis Feel free to contribute it here. There isn't any vagrant content here yet becasue I know a lot of people that are doing extension testing are using Rasmus' box, and I need to get feedback on the Windows box I built but haven't had the time to find a host to share it via.

**Comment by cebe** [comment]
@svpernova09 what kind of host do you need?

**Comment by svpernova09** [comment]
@cebe File storage. I have a 7.6G Windows Vagrant box that needs a place to live so I can set it up via vagrant cloud.

**Comment by cebe** [comment]
@svpernova09 I can give you an ftp account on my server if you want. can you ping me on IRC? "CeBe" on freenode

**Comment by rlerdorf** [comment]
7.6G?? Is there no way to slim that down a bit? The php7dev image is 1.6G right now.
Have you tried compacting it?
I do this before I export it:
inside the vm to zero out all free space: 
```
dd if=/dev/zero of=file
rm file
```
Of course, I also remove all large log files and anything not needed. Then I shut it down and do this right to the raw Virtualbox vmdk file:
```
VBoxManage clonehd box-disk1.vmdk cloned.vdi --format vdi
VBoxManage clonehd cloned.vdi box-disk2.vmdk --format vmdk
rm box-disk1.vmdk cloned.vdi
```
And in Virtualbox, you then need to click on *Storage* in settings and point it to this new compacted *box-disk2.vmdk* file. Of course, the *dd* step won't work exactly like that in Windows, but there should be some Windows equivalent for zero'ing out all unused space.

**Comment by svpernova09** [comment]
@rlerdorf I am FAR from an expert on Windows. I'm just trying to pitch in. If there is someone who is a Windows expert out there that can help I'm all for it.

**Comment by rlerdorf** [comment]
It looks like this tool is the Windows equivalent of my dd command:

https://technet.microsoft.com/en-gb/sysinternals/bb897443.aspx

So you would `sdelete -z c:`

**Comment by svpernova09** [comment]
@rlerdorf Thanks, sdelete only brought the size down to 7.3G. I'm certain it's the Visual Studio install eating up a lot of space. I just don't have the powershell skills to write scripts to do the install / configuration.

**Comment by cebe** [comment]
The image can be found on http://gophp7-ext.cebe.cc/ btw.

**Comment by auroraeosrose** [comment]
Windows is fat and visual studio is fatter - but you should be able to NOT install a lot of visual studio, since we only need the c++ bits

**Comment by svpernova09** [comment]
I've added a Development platforms section to the website and wiki, if anyone has any vagrants / dockers / vms / other they want to have listed please open a pull request or ping me (joepferguson) in #gophp7-ext on the Freenode IRC network.
